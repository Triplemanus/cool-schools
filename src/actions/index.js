export const getAllSchools = schools => ({
  type: 'GET_SCHOOLS',
  schools
});

export const searchSchools = schools => ({
  type: 'SEARCH_SCHOOLS',
  schools
});

export const addSchools = schools => ({
  type: 'ADD_SCHOOLS',
  schools
});

export const selectSchools = schools => ({
  type: 'SELECT_SCHOOLS',
  schools
});

export const displaySchools = schools => ({
  type: 'DISPLAY_SCHOOLS',
  schools
});

export const hasErrored = errorMsg => ({
  type: 'HAS_ERRORED',
  errorMsg
});

 export const isLoading = isLoading => ({
   type: 'IS_LOADING',
   isLoading
 });