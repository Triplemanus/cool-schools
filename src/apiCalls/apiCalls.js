export const fetchAllSchools = async () => {
  const url = 'https://api.schooldigger.com/v1.2/schools?st=CO&level=High&nearLatitude=39.7506294&nearLongitude=-104.9966207&distanceMiles=20&appID=d04b0481&appKey=0c80520db8adfa4e8be1ea61724e1f24';
  const response = await fetch(url);
  if (!response.ok) {
    throw new Error ('There was an error getting your data.');
  }
  const schools = await response.json();
  return schools
}

