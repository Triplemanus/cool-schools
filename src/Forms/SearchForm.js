import React, {Component } from 'react';
import { connect } from 'react-redux';
import { addSchools, searchSchools, isLoading, hasErrored } from '../actions/index';
import { fetchAllSchools } from '../apiCalls/apiCalls';

export class SearchForm extends Component {
  constructor() {
    super();
    this.state = {
      locState: '',
      level: '',
      latLocation: '',
      longLocation: '',
      maxDistance: ''
    }
  };

  updateForm = e => {
    this.setState({ [e.target.name]: [e.target.value] });
    console.log('updateForm state, locState is:', this.state, this.state.locState);
  };

  clearInputs = () => {
    this.setState({ 
      locState: '',
      level: '',
      latLocation: '',
      longLocation: '',
      maxDistance: ''
    });
  };

  handleSubmit = e => {
    e.preventDefault();
    const querySchools = {
      state: this.state.locState,
      level: this.state.level,
      latLocation: parseFloat(this.state.latLocation),
      longLocation:  parseFloat(this.state.longLocation),
      maxDistance: parseInt(this.state.maxDistance)
    };
    console.log('querySchools: ', querySchools);
    this.searchSchools(querySchools);
    this.clearInputs();

  }

  searchSchools = querySchools => {
    fetchAllSchools(querySchools)
    .then(schools => {
      console.log('schools: ', schools);
      this.props.addSchools(schools)})
    .catch(error => this.setState({ error: error }));
  }

  // searchSchools = async e => {
  //   const { isLoading, hasErrored, selectSchools } = this.props;
  //   e.preventDefault();
  //   try {
  //     isLoading(true);
  //     const newSearch = await searchSchools({ ...this.state});
  //     selectSchools({ ...this.state});
  //     isLoading(false);
  //     this.clearInputs;
  //   } catch({ message }) {
  //     isLoading(false);
  //     hasErrored(message)
  //   }
  // }

  render() {
    const { locState, level, latLocation, longLocation, maxDistance } = this.state;
    console.log('state is : ', this.state);
    return(
      <header>
        <h1>Search Form</h1>
        <form id="SearchForm">
          <input
            type="text"
            name="locState"
            value={locState}
            placeholder="State"
            onChange={this.updateForm}        
          />
          <input
            name="level"
            value={this.state.level}
            placeholder="Level"
            onChange={this.updateForm}          
          />
          <input
          name="latLocation"
          value={this.state.latLocation}
          placeholder="Latitude"
          onChange={this.updateForm}
          />
          <input
          name="longLocation"
          value={this.state.longLocation}
          placeholder="Longitude"
          onChange={this.updateForm}
          />
          <input
          name="maxDistance"
          value={this.state.maxDistance}
          placeholder="Distance"
          onChange={this.updateForm}
          />
          <button onClick={event => this.handleSubmit(event)}>
            Search Schools
          </button>
        </form>
      </header>
    )
  }
}

export const mapStateToProps = state => ({
  ...state
});

export const mapDispatchToProps = dispatch => ({
  addSchools: schools => dispatch(addSchools(schools))
});


export default connect(mapStateToProps, mapDispatchToProps)(SearchForm);
