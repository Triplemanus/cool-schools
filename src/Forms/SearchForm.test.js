import React from 'react';
import { shallow } from 'enzyme';
import { SearchForm, mapStateToProps, mapDispatchToProps } from './SearchForm';
import { fetchAllSchools } from '../apiCalls/apiCalls';
import { searchSchools, isLoading, hasErrored } from '../actions';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('SearchForm', () => {
  let wrapper, mockLocState, mockLevel, mockLatLocation, mockLongLocation, mockMaxDistance;

  beforeEach(() => {
    mockLocState = '';
    mockLevel =  '';
    mockLatLocation = '';
    mockLongLocation = '';
    mockMaxDistance = '';
    wrapper = shallow(<SearchForm 
      locState={mockLocState}
      level={mockLevel}
      latLocation={mockLatLocation}
      longLocation={mockLongLocation}
      maxDistance={mockMaxDistance}
    />);
  });

  it('should match the snapshot with all data passed in correctly', () => {

    expect(wrapper).toMatchSnapshot();
  });
});