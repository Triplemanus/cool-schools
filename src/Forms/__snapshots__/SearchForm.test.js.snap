// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot with all data passed in correctly 1`] = `undefined`;
